pkg/model/components/kubedns.go

```diff
@@ -74,5 +74,33 @@ func (b *KubeDnsOptionsBuilder) BuildOptions(o interface{}) error {
 		clusterSpec.KubeDNS.MemoryLimit = &defaultMemoryLimit
 	}
 
+	NodeLocalDNS := clusterSpec.KubeDNS.NodeLocalDNS
+	if NodeLocalDNS == nil {
+		NodeLocalDNS = &kops.NodeLocalDNSConfig{}
+		NodeLocalDNS.Enabled = false
+	} else if NodeLocalDNS.Enabled {
+		// https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/#configuration
+		NodeLocalDNS.Domain = clusterSpec.ClusterDNSDomain
+
+		if NodeLocalDNS.LocalIP == "" {
```

Comment:
These tests are usually done inside of the templates so that the spec is close to immutable inside of kops. i.e the spec is the same as the one the users set.

Reply:
Gotcha. It makes sense.